### ISSUE TITLE
Add support in poser command for ddev's new "drupal" project type

### ISSUE THREAD

**Issue by kekkis** [ddev/ddev-drupal-contrib#44]
At the moment, poser relies on ddev's `$DDEV_PROJECT_TYPE` which, beginning [DDEV v1.23.0](https://github.com/ddev/ddev/releases/tag/v1.23.0) will always be just `drupal` instead of the previous versions' `drupal8`, `drupal9`, or `drupal10`.

I'm not sure which would be the best way to get around this change but something needs to change in poser for it to work properly again. At the moment at least for me personally `ddev poser` seems to install Drupal 9 by default.

(Relevant bit of code in `.ddev/commands/web/expand-composer-json`:
```bash
[[ $DDEV_PROJECT_TYPE == "drupal10" ]] && export _TARGET_CORE=^10
[[ $DDEV_PROJECT_TYPE == "drupal9" ]] && export _TARGET_CORE=^9
[[ $DDEV_PROJECT_TYPE == "drupal8" ]] && export _TARGET_CORE=^8
```

**Comment by kekkis** [comment]
Actually I noticed another issue: the script uses the environment variable `DRUPAL_CORE` and expand-composer-json sets `_TARGET_CORE`.

See https://git.drupalcode.org/project/gitlab_templates/-/raw/default-ref/scripts/expand_composer_json.php.

**Comment by tyler36** [comment]
We should also factor in the upcoming Drupal 11 release, which DDEV supports.

**Comment by HeikkiYlipaavalniemi** [comment]
I think there might also be a larger refactoring requirement so that you could run multiple core version phpunit tests at the same time. You might have a contrib module that still supports even Drupal 8 and at least Drupal 9. And in the future there most certainly will be modules that support both D10 and D11 and we want to run the unit tests against both core versions. I think this is not at the moment possible if the core requirement is fixed to a single core in Composer

**Comment by kekkis** [comment]
> We should also factor in the upcoming Drupal 11 release, which DDEV supports.

@tyler36, I think it might be time to rethink how the version is determined. Here it is inferred from the project type but in the future this will be impossible. Scripts are able to accept user input in one way or another though, so maybe accepting an argument or even prompting for (even minor) version of Drupal might be the way to go.

**Comment by mscieszka** [comment]
This issue seems like a duplicate of #38

**Comment by HeikkiYlipaavalniemi** [comment]
@mscieszka No, I don't think it directly is. #38 is related to having D11 support which is what @tyler36 mentioned. But currently the latest beta version of the tool is broken and seems to install Drupal core 9.5 with both DDEV 1.22.7 and 1.23.0.

Basically if I setup DDEV like this:
```
ddev config --project-name=piwik-pro --project-type=drupal10 --docroot=web  --php-version=8.1
```
It installs `drupal10` project type but the poser command installs Drupal 9 core.

It is fixed by modifying `expand-composer-json` to determine the core correctly. This seems to work with both versions:
```
#!/bin/bash

#ddev-generated
## Command provided by https://github.com/ddev/ddev-drupal-contrib
## Description: Add Drupal core and other needed dependencies.
## Usage: expand-composer-json [flags] [PROJECT_NAME]
## Example: "ddev expand-composer-json ctools"
## ProjectTypes: drupal8,drupal9,drupal10
## ExecRaw: true

export _WEB_ROOT=$DDEV_DOCROOT
[[ $DDEV_PROJECT_TYPE == "drupal" ]] && export DRUPAL_CORE=^10
[[ $DDEV_PROJECT_TYPE == "drupal10" ]] && export DRUPAL_CORE=^10
[[ $DDEV_PROJECT_TYPE == "drupal9" ]] && export DRUPAL_CORE=^9
[[ $DDEV_PROJECT_TYPE == "drupal8" ]] && export DRUPAL_CORE=^8
cd "$DDEV_COMPOSER_ROOT" || exit
curl -OL https://git.drupalcode.org/project/gitlab_templates/-/raw/default-ref/scripts/expand_composer_json.php
php expand_composer_json.php "$DDEV_SITENAME"
rm -f expand_composer_json.php
```

**Comment by tyler36** [comment]
#38 is specifically for D11 support.

This (#44) is for DDEV `1.23` which removed version-specific `projectType` (Drupal 7, 8, 9 ...) in favor of `drupal` (which detects the required version).

I mentioned D11 support above because this issue will require a refactor of how it determines the correct Drupal version, and we should be considering how best to reduce maintainablity.

**Comment by kekkis** [comment]
Thanks @mscieszka, indeed there is at least significant overlap between these two. Maybe the point of view was originally a bit different and I did not even read through the discussion on the linked issue.

However the point still stands about determining the version IMO. Not sure how to go about this, should we try to separate the two issues more significantly or just try to handle everything on one of these?

**Comment by tyler36** [comment]
@HeikkiYlipaavalniemi  

We also need to update the `ProjectTypes` line. 

```diff
- ## ProjectTypes: drupal8,drupal9,drupal10
+ ## ProjectTypes: drupal8,drupal9,drupal10,drupal
```

**Comment by HeikkiYlipaavalniemi** [comment]
Updated the original comment. 

Also a small detail but the project readme currently mentions this
```
ddev config --project-name=[contrib module] --project-type=drupal10 --docroot=web --create-docroot --php-version=8.1
```

But the `--create-docroot` flag is not required anymore.

**Comment by weitzman** [comment]
Fixed in #46
